### ISSUE TITLE
Support semver version numbers when loading a plugin

### ISSUE THREAD

**Issue by hsiang9431-amzn** [opensearch-project/OpenSearch#622]
**Is your feature request related to a problem? Please describe.**
Current plugin installer fails when opensearch version is not exactly separated by `.` or `-` into 3 sections, e.g. `1.0.0`
 or `1.0-beta`. This behavior causes version string such `1.0.0-beta1` error out during installation.

**Describe the solution you'd like**
version string should not be validated with existing logic. We should rather utilize regular expression for such use case

**Additional context**
Code changes should be around here:
https://github.com/opensearch-project/OpenSearch/blob/13aa16989619e201d74f8b17beedcfe2ba264f05/server/src/main/java/org/opensearch/Version.java#L177

```java
String[] parts = version.split("[.-]");
// todo: add back optional build number
if (parts.length != 3) {
    throw new IllegalArgumentException("the version needs to contain major, minor, and revision" + version);
}
```
The code is slicing version string by `-` or `.` and checks if it ends up in 3 parts

Better logic should be used, like a regular expression
```
^v?([0-9]+\.){0,2}([0-9]+)((\-alpha[0-9]*)|(\-beta[0-9]*))?$
```

**Comment by harold-wang** [comment]
@hsiang9431-amzn thanks for reporting this. Opensearch Version info is made up of 3 portion major, minor, and revision separated by charactor '.' or '-'.  I could not tell if "Current plugin installer fails " is caused by this. Can you please post your error log? thanks, @harold-wang

**Comment by cliu123** [comment]
> @hsiang9431-amzn thanks for reporting this. Opensearch Version info is made up of 3 portion major, minor, and revision separated by charactor '.' or '-'. I could not tell if "Current plugin installer fails " is caused by this. Can you please post your error log? thanks, @harold-wang

This is the error log:
`root@70732b40d8c3:/docker-host/opensearch-1.0.0-beta1-SNAPSHOT# bin/opensearch-plugin install file:///docker-host/plugins/opensearch/
opensearch-security-1.0.0.0-beta1.zip
-> Installing file:///docker-host/plugins/opensearch/opensearch-security-1.0.0.0-beta1.zip
-> Downloading file:///docker-host/plugins/opensearch/opensearch-security-1.0.0.0-beta1.zip
[=================================================] 100%??
-> Failed installing file:///docker-host/plugins/opensearch/opensearch-security-1.0.0.0-beta1.zip
-> Rolling back file:///docker-host/plugins/opensearch/opensearch-security-1.0.0.0-beta1.zip
-> Rolled back file:///docker-host/plugins/opensearch/opensearch-security-1.0.0.0-beta1.zip
Exception in thread "main" java.lang.IllegalArgumentException: the version needs to contain major, minor, and revision1.0.0-beta1
	at org.opensearch.Version.fromStringSlow(Version.java:180)
	at org.opensearch.Version.fromString(Version.java:170)
	at org.opensearch.plugins.PluginInfo.readFromProperties(PluginInfo.java:197)
	at org.opensearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:797)
	at org.opensearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:856)
	at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:263)
	at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:237)
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:99)
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140)
	at org.opensearch.cli.MultiCommand.execute(MultiCommand.java:104)
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140)
	at org.opensearch.cli.Command.main(Command.java:103)
	at org.opensearch.plugins.PluginCli.main(PluginCli.java:60)`

**Comment by harold-wang** [comment]
@cliu123 thanks for the log, we will work on a fix and update.

**Comment by tlfeng** [comment]
@hsiang9431-amzn Thanks for pointing out the source of the error!
There was a recent change for that part of logic and seems caused the error, in this commit https://github.com/opensearch-project/OpenSearch/commit/6e04778d0c8aa33b3360bf6b000f85da93cec650.

**Comment by rursprung** [comment]
i would propose that any version check is fully compatible with [semver](https://semver.org/) as this is what opensearch will be using. so it shouldn't have any notion of knowing about `alpha`, `beta`, etc. (just saw that in the linked commit) and the test coverage should check for general semver compliance.
more likely: you can just pull in an existing semver compliant version checker library? (a random online search found this one which may or may not be something good, i spent a full 10sec looking at the README: https://github.com/kherge/java.semver)

**Comment by dblock** [comment]
💯 to @rursprung's semver suggestion, I think we possibly made a mistake introducing the alpha or beta labels into the actual version numbers.

**Comment by reta** [comment]
@nknize should we close this one? thanks!

**Comment by nknize** [comment]
fixed by #1112
